### ISSUE TITLE
Broken dependency tree with v5.0.0

### ISSUE THREAD

**Issue by samox73** [pulumi/pulumi-aws#1877]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

When creating a new pulumi project with typescript, `npm` fails to resolve dependencies:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: xxxxxx@undefined
npm ERR! Found: @pulumi/aws@5.0.0
npm ERR! node_modules/@pulumi/aws
npm ERR!   @pulumi/aws@"^5.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @pulumi/aws@"^4.0.0" from @pulumi/awsx@0.30.0
npm ERR! node_modules/@pulumi/awsx
npm ERR!   @pulumi/awsx@"^0.30.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/xxxxxx/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/xxxxxx/.npm/_logs/2022-03-24T14_51_12_019Z-debug-0.log
error: npm install failed; rerun manually to try again, then run 'pulumi up' to perform an initial deployment: exit status 1
```

Does this bug originate from `@pulumi/awsx` rather than `@pulumi/aws`?

Manually downgrading to:
```
$ cat package.json
{
    "name": "monitoring-architecture",
    "devDependencies": {
        "@types/node": "^14"
    },
    "dependencies": {
        "@pulumi/aws": "^4.0.0",
        "@pulumi/awsx": "^0.30.0",
        "@pulumi/pulumi": "^3.0.0"
    }
}
```
resolves this issue.

### Steps to reproduce

1. `pulumi new aws-typescript`

Expected: No error
Actual: Error

**Comment by stack72** [comment]
Hi @samox73 

I'm investigating this right now

Paul

**Comment by stack72** [comment]
Hi @stack72 

We shipped v0.40.0 of the awsx provider and then updated the provider - https://github.com/pulumi/templates/pull/257

This will fix fix the problem

Paul
